manifests/server.pp

```diff
@@ -89,79 +91,80 @@
+  Optional[Stdlib::Filemode]                   $postgresql_conf_mode         = $postgresql::params::postgresql_conf_mode,
+  String[1]                                    $recovery_conf_path           = $postgresql::params::recovery_conf_path,
+
+  String[1]                                    $datadir                      = $postgresql::params::datadir,
+  Optional[String[1]]                          $xlogdir                      = $postgresql::params::xlogdir,
+  Optional[String[1]]                          $logdir                       = $postgresql::params::logdir,
+
+  Optional[String[1]]                          $log_line_prefix              = $postgresql::params::log_line_prefix,
+
+  Boolean                                      $pg_hba_conf_defaults         = $postgresql::params::pg_hba_conf_defaults,
+
+  String[1]                                    $user                         = $postgresql::params::user,
+  String[1]                                    $group                        = $postgresql::params::group,
+
+  Boolean                                      $needs_initdb                 = $postgresql::params::needs_initdb,
+
+  Optional[String[1]]                          $encoding                     = $postgresql::params::encoding,
+  Optional[String[1]]                          $locale                       = $postgresql::params::locale,
+  Optional[String[1]]                          $data_checksums               = $postgresql::params::data_checksums,
+  Optional[String[1]]                          $timezone                     = $postgresql::params::timezone,
+
+  Boolean                                      $manage_pg_hba_conf           = $postgresql::params::manage_pg_hba_conf,
+  Boolean                                      $manage_pg_ident_conf         = $postgresql::params::manage_pg_ident_conf,
+  Boolean                                      $manage_recovery_conf         = $postgresql::params::manage_recovery_conf,
+  Boolean                                      $manage_postgresql_conf_perms = $postgresql::params::manage_postgresql_conf_perms,
+  Boolean                                      $manage_selinux               = $postgresql::params::manage_selinux,
+  String[1]                                    $module_workdir               = $postgresql::params::module_workdir,
+
+  Boolean                                      $manage_datadir               = $postgresql::params::manage_datadir,
+  Boolean                                      $manage_logdir                = $postgresql::params::manage_logdir,
+  Boolean                                      $manage_xlogdir               = $postgresql::params::manage_xlogdir,
+  Optional[String]                             $password_encryption          = $postgresql::params::password_encryption,
+  String                                       $extra_systemd_config         = $postgresql::params::extra_systemd_config,
```

Comment:
Can be a string of minimal length 1 String[1]

Reply:
Isn't this `Optional[String[1]]`?

Reply:
@ekohl I just transfered your suggestion from my PR. Thanks for pointing this out

Reply:
Will double check this value when I get the chance. I might have misjudged the parameter.

Reply:
Right, double checked extra_systemd_config and it looks like its defined initially at params.pp with a value of ''.
So it cannot be setup as Optional because its technically not an undef value and it doesnt get assigned an undef value either. And it cannot be a String[1] because it is indeed an empty-string.

Im not quite sure this is a change that can be implemented without breaking something else.

Reply:
I already set   $extra_systemd_config  = undef in my commit. please rebase.